### PR TITLE
1945 slider tests

### PR DESCRIPTION
**Pull request by TamMinh87**
Angular Slider Test
**What has been done**
1. Added more IDs to angular test site
2. Extended _com.epam.jdi.light.ui.html.elements.common.Range_ and override methods
3. Added more methods for Angular slider only
4. Added all related test (except Show Interval which does not has element on DOM)

### CODE REVIEW

**Comment by mf-code** [comment]
[5688] - Tkwum2LW

**Comment by mf-code** [comment]
[5688] - Pgke6ZtA

**Comment by mf-code** [comment]
[5688] - openjdk8 - Allure report on Netlify: https://5ef9cb7a60043f94ab427d79--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5688] - openjdk10 - Allure report on Netlify: https://5ef9cbdf436a66ce59293bb3--musing-swartz-6fc060.netlify.app

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    /**
+     * move slider to left
+     * @return
+     */
+    public void moveLeft() {
+        this.uiElement.sendKeys(Keys.ARROW_LEFT);
+    }
+
+    // tick interval: ignore for now
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java:151]: Please remove this comment

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    public void moveRight() {
+        this.uiElement.sendKeys(Keys.ARROW_RIGHT);
+    }
+
+    /**
+     * move slider to left
+     * @return
+     */
+    public void moveLeft() {
+        this.uiElement.sendKeys(Keys.ARROW_LEFT);
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java:148]: Replace `this.uiElement `with just `uiElement`  here and in other places in this class

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+        UIElement sliderTrackBackground = this.uiElement.find(".mat-slider-track-background");
+        UIElement sliderTrackFill = this.uiElement.find(".mat-slider-track-fill");
+        UIElement sliderThumbContainer = this.uiElement.find(".mat-slider-thumb-container");
+
+        // set value for slider attributes to reflect changes on UI
+        reflectSliderChangeOnUI(value, sliderTrackBackground, sliderTrackFill, sliderThumbContainer);
+    }
+
+    private void reflectSliderChangeOnUI(double value, UIElement sliderTrackBackground, UIElement sliderTrackFill, UIElement sliderThumbContainer) {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java:43]: Add JavaDoc

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+import com.epam.jdi.light.elements.common.UIElement;
+import org.openqa.selenium.Keys;
+import java.text.DecimalFormat;
+
+import static com.epam.jdi.light.ui.html.HtmlUtils.getDouble;
+
+/**
+ * Angular slider documentation: TBD
+ */
+public class Slider extends com.epam.jdi.light.ui.html.elements.common.Range {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java:12]: Import Range

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
+package com.epam.jdi.angular.elements.common;
+
+import com.epam.jdi.light.elements.common.UIElement;
+import org.openqa.selenium.Keys;
+import java.text.DecimalFormat;
+
+import static com.epam.jdi.light.ui.html.HtmlUtils.getDouble;
+
+/**
+ * Angular slider documentation: TBD
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java:10]: Add a link to slider documentation

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+        thumbLabelSlider.slider.setupValue(650);
+        assertEquals(thumbLabelSlider.slider.value(), 650.0);
+
+        configurableSlider.slider.show();
+        configurableSlider.slider.setupValue(35.5);
+        assertEquals(configurableSlider.slider.value(), 35.5);
+    }
+
+    @Test
+    public void test() {
```
Comment (AkioFaris) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java:127]: remove this test

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+import io.github.epam.TestsInit;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static io.github.com.StaticSite.angularPage;
+import static io.github.com.pages.AngularPage.*;
+import static io.github.epam.site.steps.States.shouldBeLoggedIn;
+import static org.testng.Assert.*;
+
+public class SliderTests extends TestsInit {
```
Comment (AkioFaris) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java:12]: Split this class into unit and UI tests accroding to the [Angular_tests_guide.docx](https://epam-my.sharepoint.com/:w:/r/personal/alexey_girin_epam_com/_layouts/15/Doc.aspx?sourcedoc=%7BE3A93BE7-D0FF-45FD-B3D2-804A31B2482C%7D&file=Angular_tests_guide.docx&action=default&mobileredirect=true)

File: jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/SliderSection.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+import com.epam.jdi.angular.elements.common.Checkbox;
+import com.epam.jdi.light.elements.common.UIElement;
+import com.epam.jdi.light.elements.composite.Section;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.UI;
+import com.epam.jdi.angular.elements.common.Slider;
+import com.epam.jdi.light.ui.html.elements.common.TextField;
+
+
+public class SliderSection extends Section {
```
Comment (AkioFaris) [inline jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/SliderSection.java:11]: SliderSection should consist of all AngularPage sliders: basic, configurable, formatting slider and the related buttons, labels, etc

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java (hunk 1 above)
Comment (tantarin) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java:151]: fixed

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java (hunk 2 above)
Comment (tantarin) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java:148]: fixed

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java (hunk 3 above)
Comment (tantarin) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java:43]: fixed

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java (hunk 4 above)
Comment (tantarin) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java:12]: fixed

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java (hunk 6 above)
Comment (tantarin) [reply jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java:127]: fixed

**Comment by mf-code** [comment]
[5693] - av7MyH9p

**Comment by mf-code** [comment]
[5693] - 0AX93QHE

**Comment by mf-code** [comment]
[5693] - xPhYwRAd

**Comment by mf-code** [comment]
[5693] - 8b7eRV4Q

**Comment by mf-code** [comment]
[5693] - QgDdpeTn

**Comment by mf-code** [comment]
[5693] - uMDrkoIG

**Comment by mf-code** [comment]
[5693] - GQrnPK49

**Comment by mf-code** [comment]
[5693] - zBPedats

**Comment by mf-code** [comment]
[5693] - uDQ91lSX

**Comment by mf-code** [comment]
[5693] - w564qkjq

**Comment by mf-code** [comment]
[5694] - 5cHgRa0S

**Comment by mf-code** [comment]
[5694] - hnATDyND

**Comment by mf-code** [comment]
[5694] - dl34t12n

**Comment by mf-code** [comment]
[5693] - ku8fEnVt

**Comment by mf-code** [comment]
[5694] - zmtsXsIK

**Comment by mf-code** [comment]
[5694] - 6oY86svd

**Comment by mf-code** [comment]
[5694] - 12L4fyf6

**Comment by mf-code** [comment]
[5694] - ZicOewpG

**Comment by mf-code** [comment]
[5694] - zJ1WqV60

**Comment by mf-code** [comment]
[5693] - LAKW2EEH

**Comment by mf-code** [comment]
[5694] - 0tpZAx20

**Comment by mf-code** [comment]
[5694] - 5CW2KAgx

**Comment by mf-code** [comment]
[5693] - openjdk8 - Allure report on Netlify: https://5efaeca590f3ba5d3bafc9d2--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5695] - B5Uh1vM3

**Comment by mf-code** [comment]
[5695] - f4z4a92Q

**Comment by mf-code** [comment]
[5693] - openjdk10 - Allure report on Netlify: https://5efaed72ade033187840e3d3--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5694] - FerDO8gM

**Comment by mf-code** [comment]
[5694] - NTNjFnfS

**Comment by mf-code** [comment]
[5695] - hszishwD

**Comment by mf-code** [comment]
[5695] - UjDPHIQ5

**Comment by mf-code** [comment]
[5695] - Or6wBNgZ

**Comment by mf-code** [comment]
[5695] - EUqVU7ZB

**Comment by mf-code** [comment]
[5694] - openjdk8 - Allure report on Netlify: https://5efaefbc4953717965d2f009--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5695] - XAb6kI8h

**Comment by mf-code** [comment]
[5694] - openjdk10 - Allure report on Netlify: https://5efaf0bbade033497140e23e--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5695] - E62IdWsc

**Comment by mf-code** [comment]
[5695] - 0dSruKc9

**Comment by mf-code** [comment]
[5695] - tQ6SZcfQ

**Comment by mf-code** [comment]
[5695] - HFlESZBw

**Comment by mf-code** [comment]
[5695] - bSWkekb7

**Comment by mf-code** [comment]
[5695] - openjdk8 - Allure report on Netlify: https://5efaf5263f9e7c3431f1a759--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5695] - openjdk10 - Allure report on Netlify: https://5efaf5f0635cb6c5010d1a2f--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5884] - 8De2xLmY0Oec

**Comment by mf-code** [comment]
[5884] - uW6kLjrvsdRB

**Comment by mf-code** [comment]
[5884] - ZjjtKjaQPnqW

**Comment by mf-code** [comment]
[5884] - DF4JtxvmNdYY

**Comment by mf-code** [comment]
[5884] - InClnYuRouuM

**Comment by mf-code** [comment]
[5884] - l9267KeM0HTr

**Comment by mf-code** [comment]
[5884] - q5dsTcLfjslV

**Comment by mf-code** [comment]
[5884] - fZMlTJ79I0OG

**Comment by mf-code** [comment]
[5884] - gdglDS4YFp32

**Comment by mf-code** [comment]
[5884] - JTY8xfZhWrAU

**Comment by mf-code** [comment]
[5884] - FX6L1xJzOoiS

**Comment by mf-code** [comment]
[5884] - DUtwBjrIWWlA

**Comment by mf-code** [comment]
[5884] - eY8eR5f1vckV

**Comment by mf-code** [comment]
[5884] - vGUiJZsXC8ZR

**Comment by mf-code** [comment]
[5884] - openjdk8 - Allure report on Netlify: https://5f354bb7f1feef015eacfcc0--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5884] - openjdk10 - Allure report on Netlify: https://5f354cab82b0ca00e76e9486--musing-swartz-6fc060.netlify.app

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+    }
+
+    @Test
+    public void minTest() {
+        sliderSection.matSliderBasic.show();
+        sliderSection.matSliderBasic.is().minValue(0.0);
+
+        sliderSection.matSliderFormatting.show();
+        sliderSection.matSliderFormatting.is().minValue(1.0);
+
```
Comment (Vtablos) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java:33]: redundant space

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+        sliderSection.sliderConfigurableMin.setValue("0");
+        sliderSection.sliderConfigurableMax.setValue("100");
+        sliderSection.sliderConfigurableStep.setValue("1");
+    }
+
+    @Test
+    public void minTest() {
+        sliderSection.matSliderBasic.show();
+        sliderSection.matSliderBasic.is().minValue(0.0);
+
```
Comment (Vtablos) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java:30]: redundant space

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+        sliderSection.matSliderConfigurable.is().minValue(0.0);
+        sliderSection.sliderConfigurableMin.setValue("10");
+        sliderSection.matSliderConfigurable.is().minValue(10.0);
+    }
+
+    @Test
+    public void maxTest() {
+        sliderSection.matSliderBasic.show();
+        sliderSection.matSliderBasic.is().maxValue(100.0);
+
```
Comment (Vtablos) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java:44]: redundant space

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+        sliderSection.matSliderFormatting.show();
+        sliderSection.matSliderFormatting.is().maxValue(100000.0);
+
```
Comment (Vtablos) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java:47]: redundant space

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+        sliderSection.matSliderConfigurable.is().maxValue(1000.0);
+    }
+
+    @Test
+    public void currentValueTest() {
+        sliderSection.matSliderConfigurable.show();
+        sliderSection.sliderConfigurableThumbLabel.check();
+        sliderSection.sliderConfigurableValue.setValue("62");
+        sliderSection.matSliderConfigurable.has().value(62.0);
+
```
Comment (Vtablos) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java:60]: ...

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+        sliderSection.sliderConfigurableValue.setValue("100");
+        sliderSection.matSliderConfigurable.has().value(100.0);
+
```
Comment (Vtablos) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java:63]: ...

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java [hunk 15]
```diff
@@ -0,0 +1,211 @@
...
+        sliderSection.sliderConfigurableValue.setValue("80.5");
+        sliderSection.matSliderConfigurable.has().value(80.5);
+
```
Comment (Vtablos) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java:66]: ...

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java [hunk 16]
```diff
@@ -0,0 +1,211 @@
...
+        sliderSection.sliderConfigurableValue.setValue("0");
+        sliderSection.matSliderConfigurable.has().value(0.0);
+    }
+
+    @Test
+    public void stepValueTest() {
+        sliderSection.matSliderConfigurable.show();
+        sliderSection.sliderConfigurableStep.setValue("5");
+        sliderSection.sliderConfigurableThumbLabel.check();
+
```
Comment (Vtablos) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SliderTests.java:76]: ....

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/common/Slider.java [hunk 17]
```diff
@@ -0,0 +1,152 @@
...
+
+    @JDIAction(value = "Is '{name}' thumb label displayed", level = DEBUG)
+    public boolean isThumbLabelDisplayed() {
+        return uiElement.find(".mat-slider-thumb-label-text").isVisible();
+    }
+
+    @JDIAction(value = "Set value '{0}' for '{name}'", level = DEBUG)
+    public void setupValue(double value) {
+        uiElement.setAttribute("aria-valuenow", value + "");
+
```
Comment (Vtablos) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/common/Slider.java:55]: redundant space

**Comment by mf-code** [comment]
[5885] - UGUovVY3ryMO

**Comment by mf-code** [comment]
[5885] - rVvtS3yBbZGZ

**Comment by mf-code** [comment]
[5885] - XqcHzjLqbb0b

**Comment by mf-code** [comment]
[5885] - TIpsnsbSSNw9

**Comment by mf-code** [comment]
[5885] - gS0vADHU4baQ

**Comment by mf-code** [comment]
[5885] - z6xLrNqvaENh

**Comment by mf-code** [comment]
[5885] - aRDDl6TjBqDi

**Comment by mf-code** [comment]
[5885] - Q7DtU1Br9Lzh

**Comment by mf-code** [comment]
[5885] - 0EeXMTYLpFko

**Comment by mf-code** [comment]
[5885] - fIuApLkcjQ4M

**Comment by mf-code** [comment]
[5885] - iFZ10bvLw1Oz

**Comment by mf-code** [comment]
[5885] - TDXg2yaKGRBF

**Comment by mf-code** [comment]
[5885] - Kz9dLRp9AwrH

**Comment by mf-code** [comment]
[5885] - GsOUQw1SATaR

**Comment by mf-code** [comment]
[5885] - openjdk8 - Allure report on Netlify: https://5f39862272903ed030781dcc--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5885] - openjdk10 - Allure report on Netlify: https://5f39871346997969d713557d--musing-swartz-6fc060.netlify.app

**Review by ViktarR — APPROVED** [review]

**Pull request merged**